### DISCUSSION TITLE
How to get started as a beginner?

### DISCUSSION THREAD

**Discussion by datascribbler777** [ocornut/imgui#5900] [New Users Build/Link/Run issues ONLY!]
Sorry to have to ask, I know the C++ community hates newbies, but I need help.
I am wanting to get started with something actually graphical with C++. This GUI seems to be highly recommended. I messed around with raylib for a bit, but that doesn't seem to support C++. 

How do I use one of the examples? 
I was looking at example_glfw_opengl3
When I try to use the makefile it fails for fatal error: GLFW/glfw3.h: No such file or directory
I poked around the directories but I can't seem to find that file.

Do I have to build this repo into some kind of static library that I then link into my .exe?

When I try to run the .bat file, it just makes a debug folder, which is empty

I am guessing the vcxproj files are some microsoft stuff, which I will not be using.
I would prefer to remain ide agnostic. 

I have tried searching for another answer, but it seems like everyone else's tutorials always rely on some tool that I do not have access to, or understand how to use. Then I spend time trying to figure that out and just get frustrated that I am not getting anywhere using this GUI.

**Comment by PathogenDavid** [comment]
> I am guessing the vcxproj files are some microsoft stuff, which I will not be using.
> I would prefer to remain ide agnostic.

IMO as a beginner you shouldn't artificially restrict yourself from using the tools your more experienced peers are using.

Early on you should not be overly opinionated about the tools you're using.

All build systems are supremely beginner-unfriendly, but Makefiles in particular are one of the worst.

> I have tried searching for another answer, but it seems like everyone else's tutorials always rely on some tool that I do not have access to, or understand how to use. Then I spend time trying to figure that out and just get frustrated that I am not getting anywhere using this GUI.

You should focus on understanding the tools first, and you should probably be focusing on simple command line apps before anything graphical.

You need to learn to walk before you can run.

**Comment by sergeyn** [comment]
- Start playing with already working examples. 
- Read books on the tools and language and operating system you are using/want to use
- Find a person or a community you can learn from and who can get your question answered or perhaps even mentor you

**Comment by datascribbler777** [comment]
I can't start "playing" with this GUI because I don't know how to get started. 
Do I have to "build" this first like I did with raylib?
Or, is this intended as a header only include?
Why does the makefile fail when I try to use an example?

**Comment by sergeyn** [comment]
- google `GLFW/glfw3.h: No such file or directory`

**Comment by ocornut** [comment]
Adding a link to the Getting Started section here for posterity, since wiki are not properly crawled:
https://github.com/ocornut/imgui/wiki/Getting-Started
